File: 125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 14/07/2026, 14:31:41
class Solution {
    public boolean isPalindrome(String s) 
    {
        s=s.toLowerCase();
        StringBuilder p=new StringBuilder();
        for(char c:s.toCharArray())
        {
            if((c>=97&&c<=122)||(c>=48&&c<=57))
            p.append(c);
        };
        int i=0;
        int j=p.length()-1;
        while(i<j)
        {
            if(p.charAt(i)!=p.charAt(j))
            {
                return false;
            }
            else
            {
                i++;
                j--;
            }
        }
        return true;   
    }
}

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 14/07/2026, 14:31:38
class Solution:
    def isPalindrome(self, s: str) -> bool:
        palindrome=[]
        for ch in s:
            if ch.isalpha() or (ord(ch) >= ord('0') and ord(ch) <= ord('9')):
                palindrome.append(ch.lower())
        i=0
        j=len(palindrome)-1
        print(j)
        while(i<j):
            if(palindrome[i]!=palindrome[j]):
                return False
            i+=1
            j-=1
        return True
        